spec/type_checker/levels/strict_spec.rb

```diff
@@ -462,6 +463,21 @@ def bar
       expect(checker.problems.first.message).to include('does not match inferred type')
     end
 
+    xit 'requires strict return tags' do
+      checker = type_checker(%(
+        class Foo
+          # The tag is [String] but the inference is [String, nil]
+          #
+          # @return [String]
+          def bar
+            true ? nil : 'bar'
+          end
+        end
+      ))
+      expect(checker.problems).to be_one
+      expect(checker.problems.first.message).to include('does not match inferred type')
+    end
```

Comment:
The above two specs don't both pass today simultaneously, which seems like an uneven user experience - I think that we should increase the quality of inference before we can bring this back into an official level so that preserving the current limited ability doesn't force us to make larger individual changes than we'd like to improve it.

Reply:
For context, leaving this in took us from 180-ish strict violations to about 500, almost all of which were false alarms.